frontend/src/app/main/data/workspace/tokens/application.cljs

```diff
@@ -55,7 +95,8 @@
                         objects (dsh/lookup-page-objects state)
 
                         shape-ids (or (->> (select-keys objects shape-ids)
-                                           (filter (fn [[_ shape]] (not= (:type shape) :group)))
+                                           (filter (fn [[_ element]]
+                                                     (seq (set/intersection attributes (shape-type->attributes (:type element))))))
```

Comment:
I'd write a helper function for this. E.g.
```clojure
common.types.token
(defn applicable-attrs
  [attributes type shape]
  (set/intersection attributes (shape-type->attributes type))
```

(and change the name `element`, it's just a shape.

Reply:
Even a second helper `any-applicable-attr?` that returns boolean.

This way you have a higher level interface, and the code in the event is easier to read.